### PR TITLE
OCM-141: Terraform for IMDSv2 support

### PR DESCRIPTION
**Pull request by tsorya**
Adding support for http_tokens_state field. This field can be used only
    from 4.11.0 and have 2 options: optional/required

### CODE REVIEW

**Comment by sagidayan** [comment]
SGTM, @bardielle Can you also take a look?

File: provider/cluster_rosa_classic_resource.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ const (
 	maxClusterNameLength = 15
 	tagsPrefix           = "rosa_"
 	tagsOpenShiftVersion = tagsPrefix + "openshift_version"
+	lowestHttpTokensVer  = "4.11.0"
```
Comment (bardielle) [inline provider/cluster_rosa_classic_resource.go:60]: What if the cluster version will be set to 4.10? (the MinVersion of cluster version is "4.10")

File: provider/cluster_rosa_classic_resource.go [hunk 2]
```diff
@@ -698,14 +715,31 @@ func (r *ClusterRosaClassicResource) getAndValidateVersionInChannelGroup(ctx con
...
+		return nil
+	}
+
+	version = getOcmVersionMinor(version)
+	a, err := ver.NewVersion(version)
 	if err != nil {
-		return fmt.Errorf("Could not get cluster version: %v", err)
+		return fmt.Errorf("version '%s' is not supported: %v", version, err)
+	}
+	b, _ := ver.NewVersion(lowestHttpTokensVer)
```
Comment (bardielle) [inline provider/cluster_rosa_classic_resource.go:728]: we already have a common logic in the function checkMinSupportedVersion
Can you reuse it? or adapting that function to be compatible with your validation?

File: provider/cluster_rosa_classic_resource.go (hunk 1 above)
Comment (bardielle) [reply provider/cluster_rosa_classic_resource.go:60]: I just thought we would like to be aligned with the minimum version 
But I think it's fine to handle it like that

**Review by bardielle — APPROVED** [review]
/lgtm

**Pull request merged**